nix/modules/tezos-node.nix

```diff
@@ -0,0 +1,119 @@
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+{config, lib, pkgs, ...}:
+
+with lib;
+
+let
+  tezos-node-pkg = pkgs.ocamlPackages.tezos-node;
+  cfg = config.services.tezos-node;
+  sources = import ../nix/sources.nix;
+  genConfigCommand = historyMode: rpcPort: netPort: network: ''
+    --data-dir "$node_data_dir" \
+    --history-mode "${historyMode}" \
+    --rpc-addr ":${toString rpcPort}" \
+    --net-addr ":${toString netPort}" \
+    --network "${network}"
+  '';
+  common = import ./common.nix { inherit lib; inherit pkgs; };
+  instanceOptions = types.submodule ( {...} : {
+    options = common.sharedOptions // {
+      enable = mkEnableOption "Tezos node service";
+
+      package = mkOption {
+        default = tezos-node-pkg;
+        type = types.package;
+      };
+
+      rpcPort = mkOption {
+        type = types.int;
+        default = 8732;
+        example = 8732;
+        description = ''
+          Tezos node RPC port.
+        '';
+      };
+
+      netPort = mkOption {
+        type = types.int;
+        default = 9732;
+        example = 9732;
+        description = ''
+          Tezos node net port.
+        '';
+      };
+
+      network = mkOption {
+        type = types.str;
+        default = "florencenet";
+        description = ''
+          Network which node will be running on.
+        '';
+      };
+
+      historyMode = mkOption {
+        type = types.str;
+        default = "full";
+        description = ''
+          Node history mode. Possible values are:
+          full, experimental-rolling or arcive.
```

Comment:
Should this use `types.enum` too?

Also `arcive` looks like a typo, should be `archive`.